digdag-core/src/main/java/io/digdag/core/storage/ArchiveManager.java

```diff
@@ -99,9 +114,17 @@ private String formatFilePath(int siteId, String projectName, String revisionNam
     {
         Preconditions.checkArgument(!Strings.isNullOrEmpty(projectName), "projectName");
         Preconditions.checkArgument(!Strings.isNullOrEmpty(revisionName), "revisionName");
+        // It's possible the storage used to store a project archive in doesn't accept some characters.
+        // For instance, AWS S3 says the following characters are generally safe
+        //   - Alphanumeric characters [0-9a-zA-Z]
+        //   - Special characters !, -, _, ., *, ', (, and )
+        //
+        // So we'd better encode a project name into the characters above
+        String encodedProjectName = Base64.getEncoder().encodeToString(projectName.getBytes(UTF_8)).replace("=", "_");
         return String.format(ENGLISH,
-                "%d/%s/%s.%s.tar.gz",
-                siteId, projectName, revisionName, DATE_TIME_SUFFIX_FORMAT.format(Instant.now()));
+                "%s%d/%s/%s.%s.tar.gz",
+                pathPrefix, siteId, encodedProjectName, revisionName,
+                DATE_TIME_SUFFIX_FORMAT.format(Instant.now()));
```

Comment:
I found a minor problem...: `ProjectResource.putProject` uploads a file to a storage, and then inserts a row to RDBMS. This INSERT may fail if the projectName & revision is already used. But the path on storage uses projectName & revision. So, if two users upload the same revision name at the same time, one fails due to unique index but the file could be overridden. `Instant.now` is added to the path to prevent this issue but its precision is second. Conflict may still happen.